src/coreclr/jit/gentree.cpp

```diff
@@ -24313,6 +24317,29 @@ bool GenTreeHWIntrinsic::OperIsMemoryLoadOrStore() const
     return OperIsMemoryLoad() || OperIsMemoryStore();
 }
 
+//------------------------------------------------------------------------------
+// OperRequiresAsgFlag : Check whether the operation requires GTF_ASG flag regardless
+//                       of the children's flags.
+//
+bool GenTreeHWIntrinsic::OperRequiresAsgFlag() const
```

Comment:
Same comment on overloading as for the liveness method.

Reply:
We have this pattern all throughout the JIT. I don't think its worth differentiating.

The debuggers can and should be resolving to the correct overload in stack traces and other scenarios since this is C++ code.

Reply:
I don't think we have other places where a callee is named the same as the caller (outside of recursion).

The comment about the stack traces is not about debuggers but people, i. e.:
```
MethodWithName
MethodWithName
CallerMethod
```
It's not as easy to read / navigate as:
```
MethodWithADifferentName
MethodWithName
CallerMethod
```
That said, in `OperRequiresAsgFlag`'s case, it will actually include the class name so it's not that relevant as for the liveness method.

Reply:
The stack traces for C++ code should typically include the full overload of the method and therefore which overload it applies to?

This should be no different than debugging C# stack traces really...

Reply:
It depends on which tool one uses, naturally...

VS does show parameters:
![image](https://user-images.githubusercontent.com/62474226/228953429-6012dc36-95b3-4dbd-83dd-3333de69d4ef.png)

WinDbg doesn't:
![image](https://user-images.githubusercontent.com/62474226/228954203-769575e1-c773-4d7c-9968-8a4bbfc4fe61.png)

Reply:
This sounds like a tooling issue that WinDBG needs to resolve and not something we should be working around in the JIT.

We ultimately have many APIs throughout the JIT which rely on either inheritance, overload resolution, or even templatization. While these aren't as extensively used as they might be in some codebases, they are very powerful tools that can help improve performance and overall usability.

C++, much like C#, is an object oriented language. Debuggers working with it need to have some handling to correctly support features like this.

Reply:
> WinDbg doesn't (by default with `k`, I am sure there is a way to make it show them):

fwiw, `kp` or `kP` should do it. I always click the `Source args` button in the `Calls` window to show them there.

Reply:
I'm very sympathetic to requests for using different names -- I do most of my source browsing using the equivalent of 'grep' and unique names can therefore be very helpful -- but the cases here seem fine and justified, to me.